telemetry/request-log/src/main/scala/io/buoyant/telemetry/requestLog/RequestLogTracer.scala

```diff
@@ -0,0 +1,99 @@
+package io.buoyant.telemetry.requestLog
+
+import com.google.common.cache.CacheBuilder
+import com.twitter.finagle.tracing._
+import com.twitter.util.Time
+import scala.collection.JavaConverters._
+import com.twitter.finagle.buoyant.Sampler
+
+class RequestLogTracer(sampleRate: Double, capacity: Int) extends Tracer {
+
+  private[this] val sampler = Sampler(sampleRate.toFloat)
+
+  private[this] val cache = CacheBuilder.newBuilder().maximumSize(capacity).build[SpanId, Seq[Record]]()
+
+  override def record(record: Record): Unit = {
+    val traceId = record.traceId.traceId
+    if (sampler(traceId.toLong) && acceptedAnnotations(record.annotation)) synchronized {
```

Comment:
How often is this being called?

Maybe we can use lock free structures?
**ConcurrencyMap** has _putIfAbsent_ and by combining with a concurrent linear structure e.g. **ConcurrentLinkedQueue**, then we should be able to remove the synchronized block?

Something like:
`
cache.putIfAbsent(traceId, new ConcurrentLinkedQueue())
cache.get(traceId).add(record)
`